cli/src/test/fake_lycan.ts

```diff
@@ -0,0 +1,85 @@
+import { TestInterface } from 'ava';
+import * as td from 'testdouble';
+import { Shell, success } from 'specshell';
+import { LycanHandler, LycanServer } from '@binaris/spice-koa-server';
+import { AddressInfo, Server } from 'net';
+import { tmpdir } from 'os';
+import { mkdir, mkdtemp , realpath, writeFile } from 'mz/fs';
+import { env as processEnv } from 'process';
+import * as path from 'path';
+import rimrafCb = require('rimraf');
+import { promisify } from 'util';
+
+const rimraf = promisify(rimrafCb);
+
+export interface Context {
+  shell: Shell;
+  run: string;
+  configDir: string;
+  configPath: string;
+  projectDir: string;
+  lycanUrl: string;
+  lycanFake: td.TestDouble<LycanHandler>;
+  lycanServer: LycanServer;
+  server: Server;
+}
+
+// Adds before and after hooks to help test with a shell against a
+// fake Lycan server.
+export function addFake<C extends Context>(test: TestInterface<C>) {
+  test.before(async (t) => {
+    // Quoted in case dirname includes spaces etc.
+    t.context.run = `'${path.resolve(__dirname, '../..', 'bin/run')}'`;
```

Comment:
Needs to be shell-escaped for paths with spaces.

Reply:
Good idea!  It's quoted, but that's obviously not nearly as good.